Branch-SDK/Branch-SDK/BNCLinkCache.m

```diff
@@ -0,0 +1,36 @@
+//
+//  BNCLinkCache.m
+//  Branch-SDK
+//
+//  Created by Qinwei Gong on 1/23/15.
+//  Copyright (c) 2015 Branch Metrics. All rights reserved.
+//
+
+#import "BNCLinkCache.h"
+
+@implementation BNCLinkCache
+
+- (id)init {
+    self = [super init];
+    if (self) {
+        self.cache = [[NSMutableDictionary alloc] init];
+    }
+    return self;
+}
+
+- (void)setObject:(NSString *)anObject forKey:(BNCLinkData *)aKey {
+    [self.cache setObject:anObject forKey:aKey];
+}
+
+- (NSString *)objectForKey:(BNCLinkData *)aKey {
+    NSUInteger hashcode = [aKey hash];
```

Comment:
Hmm, tbh I guess I'm not sure. It seems like that should work to me, but I haven't had a chance to try it yet. However, if you instead just keyed on the hash value (`@([aKey hash])`), you could avoid the need for that check entirely (and it would be O(1) vs O(n)).